Feat/legal posture (#68)
edgee-cloud/docs: getting-started/component-configuration.mdx

---
title: Component Configuration
description: Configure your first components.
icon: square-4
---

import EdgeeSdk from '/snippets/edgee-sdk.mdx';

<EdgeeSdk />

> Components are the building blocks of Edgee's functionality. Each component can be configured to meet your specific needs while maintaining privacy and compliance requirements. This guide will help you understand how to configure components effectively.

Reviewer: ```suggestion
Components are the building blocks of Edgee's functionality. You can configure each component to meet your specific needs while maintaining privacy and compliance requirements. This guide will help you configure your first component effectively.
```

<Frame caption="Choose the component you want to configure.">
  <img
    src="/images/component-catalog.png"
    alt="Choose the component you want to configure."
    align={"center"}
  />
</Frame>

## Basic Configuration

Every component in Edgee has a set of common configuration options that control its behavior. These options are available in the component configuration form:

<Frame caption="Configure your component.">
  <img
    src="/images/google-analytics-component.png"
    alt="Configure your component."
    align={"center"}
  />
</Frame>

### Common Settings

All components share these basic configuration options:

1. **Component Status**: Enable/disable the component
2. **Auto-update**: Enable/disable the component auto-update
    - `On`: Automatically update the component version when a new version is available
    - `Off`: Manually update the component version

### Data Collection specific settings

1. **Event Activation**
    - **Page Events**: Enable/disable page view tracking
    - **Track Events**: Enable/disable custom event tracking
    - **User Events**: Enable/disable user identification

> 2. **Anonymization**: Controls whether data is anonymized when consent is pending or denied

Reviewer: Do we mention anywhere else that anonymization only applies when consent is pending or denied? 🤔
